nucliadb_node/src/shards/indexes.rs

```diff
@@ -131,8 +132,11 @@ impl ShardIndexesFile {
     }
 
     pub fn store(&self, shard_path: &Path) -> NodeResult<()> {
-        let mut writer = BufWriter::new(File::create(shard_path.join(SHARD_INDEXES_FILENAME))?);
+        let filename = shard_path.join(SHARD_INDEXES_FILENAME);
+        let temp = shard_path.join(TEMP_SHARD_INDEXES_FILENAME);
+        let mut writer = BufWriter::new(File::create(temp.clone())?);
         serde_json::to_writer(&mut writer, &self)?;
+        std::fs::rename(temp, filename)?;
```

Comment:
I think you need to flush the writer before the rename in order for things to be 100% consistent.

As it is now, between the rename and the BufWritter dropping out of scope (when it flushes automatically), the file can be empty.

Reply:
Oh, thanks! I'll add i